src/commands/moderation/repel.ts

```diff
@@ -0,0 +1,402 @@
+
+const DEFAULT_LOOK_BACK = 10 * MINUTE;
+const DEFAULT_TIMEOUT_DURATION = 1 * HOUR;
+
+const isUserInServer = (target: User | GuildMember): target is GuildMember => {
+  return target instanceof GuildMember;
+};
+
+const isUserTimedOut = (target: GuildMember) => {
+  return target.communicationDisabledUntilTimestamp
+    ? target.communicationDisabledUntilTimestamp > Date.now()
+    : false;
+};
+
+const checkCanRepel = ({
+  commandUser,
+  repelRole,
+}: {
+  commandUser: GuildMember;
+  repelRole: Role;
+}): {
+  ok: boolean;
+  message?: string;
+} => {
+  const hasPermission =
+    commandUser.permissions.has(PermissionFlagsBits.ModerateMembers) ||
+    commandUser.roles.cache.has(repelRole.id);
+  if (!hasPermission) {
+    return { ok: false, message: 'You do not have permission to use this command.' };
+  }
+  return { ok: true };
+};
+
+const checkCanRepelTarget = ({
+  target,
+  commandUser,
+  botMember,
+}: {
+  target: User | GuildMember;
+  commandUser: GuildMember;
+  botMember: GuildMember;
+}): {
+  ok: boolean;
+  message?: string;
+} => {
+  if (!isUserInServer(target)) {
+    return { ok: true };
+  }
+
+  if (target.user.bot) {
+    return { ok: false, message: 'You cannot repel a bot.' };
+  }
+
+  if (target.id === commandUser.id) {
+    return { ok: false, message: 'You cannot repel yourself.' };
+  }
+
+  const isTargetServerOwner = target.id === target.guild.ownerId;
+  if (isTargetServerOwner) {
+    return { ok: false, message: 'You cannot repel the server owner.' };
+  }
+
+  if (target.roles.highest.position >= commandUser.roles.highest.position) {
+    return { ok: false, message: 'You cannot repel this user due to role hierarchy.' };
+  }
+
+  if (target.roles.highest.position >= botMember.roles.highest.position) {
+    return { ok: false, message: 'I cannot repel this user due to role hierarchy.' };
+  }
+
+  return { ok: true };
+};
+
+const getTargetFromInteraction = async (
+  interaction: ChatInputCommandInteraction
+): Promise<GuildMember | User> => {
+  const targetFromOption = interaction.options.getUser(RepelOptions.TARGET, true);
+  let target: User | GuildMember | null = null;
+  if (!interaction.inGuild() || interaction.guild === null) {
+    return targetFromOption;
+  }
+  try {
+    target = await interaction.guild.members.fetch(targetFromOption.id);
+  } catch {
+    target = targetFromOption;
+  }
+  return target;
+};
+
+const handleTimeout = async ({
+  target,
+  duration,
+}: {
+  target: GuildMember | User;
+  duration: number;
+}): Promise<number> => {
+  if (duration === 0 || !isUserInServer(target) || isUserTimedOut(target)) {
+    return 0;
+  }
+  try {
+    await target.timeout(duration * HOUR, 'Repel command executed');
+    return duration;
+  } catch {
+    return 0;
+  }
+};
+
+const getTextChannels = (interaction: ChatInputCommandInteraction) => {
+  if (!interaction.inGuild() || !interaction.guild) {
+    return [];
```

Comment:
when could this happen? maybe add console.error here as well. maybe something like
```ts
console.error(`Interaction not in a guild: ${interaction}`)
```